activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -197,9 +197,15 @@ def begin_db_transaction
           execute "BEGIN"
         end
 
-        def begin_isolated_db_transaction(isolation)
+        def begin_isolated_db_transaction(opts)
```

Comment:
We should probably use keyword arguments here, the code can be cleaned up significantly and we'll be giving the user a much nicer error message.